x-pack/plugins/cloud_security_posture/public/pages/rules/index.tsx

```diff
@@ -63,7 +64,9 @@ export const Rules = ({ match: { params } }: RouteComponentProps<PageUrlParams>)
         }
       />
       <EuiSpacer />
-      <RulesContainer />
+      <RulesProvider>
+        <RulesContainer />
+      </RulesProvider>
```

Comment:
Since `RulesProvider` is only used internally within the `RulesContainer` subtree, I wonder if there's a way to keep these lines as they were so we can encapsulate the usage of the provider. `RulesContainer` could render the provider wrapping the existing children.

That would imply `const { rulesFlyoutData } = useRules();` to be run from within `RuleFlyout` rather than from `RulesContainer`. Then `RuleFlyout` would call the hook, and if `rulesFlyoutData.metadata` is not truthy, it would render null.

Do you think that makes sense?

Reply:
Rules Flyout:  I agree,  the `RulesFlyout` can have some more work, for example it should know whether to render itself or not,  rather than rely on the parent to decide if it should be rendered if `rulesFlyoutData` data exists.

RulesProvider: Since the `RulesContainer` is now almost an empty shell,  I was consider if this proposal is accepted to most the container code directly to the page.  Moving the RulesProvider to the RulesContainer make sense though.